src/coreclr/vm/stublink.cpp

```diff
@@ -2128,7 +2132,9 @@ Stub* Stub::NewStub(PTR_VOID pCode, DWORD flags)
     }
 
     size_t stubPayloadOffset = totalSize -
-        (sizeof(Stub) + ((flags & NEWSTUB_FL_EXTERNAL) ? sizeof(PTR_PCODE) : numCodeBytes));
+        (sizeof(Stub)
```

Comment:
> Rather than try to compute backwards perhaps we should just set stubPayloadOffset to 0 initially and change it to the size of the unwind data if there is any?

Yep. This is confusing, but your table above was helpful. Adding it as a comment.